### PR TITLE
Add Warning Suppressing Effect

### PR DESCRIPTION
**Pull request by JRoy**
*before submitting, remove all help text like this*

Target Minecraft versions: N/A
Requirements: N/A
Related issues: #1260 

Description:
This pull request adds script-specific warning suppression for the following warnings:
* Variable Conflict
* Variable will not be saved
* List missing conjunctions
* Starting Variable with expression

This is accomplished through EffSuppressWarnings:
`[local[ly]] suppress (1¦conflict|2¦variable save|3¦[missing] conjunction[s]|4¦starting [with] expression[s]) warning[s]`

This effect uses the ScriptOptions class to manage what scripts are suppressing what errors.

### CODE REVIEW

**Comment by Pikachu920** [comment]
I feel like local disable should be default behavior

**Comment by bensku** [comment]
Don't suppress potential nullness issues when there are alternative ways to solve them available. It might cause issues in future if you do it method or class level, even when your usage is technically safe.

**Comment by JRoy** [comment]
@bensku, I'll just remove them, they appear to only be ide issues. So if I remove the suppresswarnings you should be fine

**Comment by Blueyescat** [comment]
Hmm i think an enum type for the warnings would be better:
```
variable [name] conflict
(starting variable [name]|variable [name] start(ing|s)) with [an] expression
(variable will not be saved|variable sav(ing|e)))
[list] [missing] [and/or] conjunction[s]
```
```autohotkey
([locally] suppress|locally disable) %warnings% warning[s]
(suppress|disable) %warnings% warning[s] for ([the] [current]|this) script
```
```vb
supress variable name conflict, variable will not be saved and variable starts with an expression warnings
```

**Comment by Pikachu920** [comment]
I agree mostly, but maybe not as an enum. Doing it like that would mean taking a different approach to this (e.g. config options for skript have an english name assigned to them during registration)

File: src/main/java/ch/njol/skript/effects/EffSuppressWarnings.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+@Name("Locally Suppress Warning")
+@Description("Suppresses target warnings from the current script.")
+@Examples({"locally suppress conflict warnings",
+			"suppress variable save warnings"})
+@Since("INSERT VERSION")
+@SuppressWarnings("null")
+public class EffSuppressWarnings extends Effect {
+
+	//[local[ly]] suppress (1¦conflict|2¦variable save|3¦[missing] conjunction[s]|4¦starting [with] expression[s]) warning[s]
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/effects/EffSuppressWarnings.java]: why is this here?

File: src/main/java/ch/njol/skript/effects/EffSuppressWarnings.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+				Skript.error("Skript returned an invalid parse mark, this should never happen. Please report this!");
+				return false;
+			}
+		}
+		return true;
+	}
+
+	@Override
+	public String toString(Event e, boolean debug) {
+		return "Locally Suppress Warning";
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/effects/EffSuppressWarnings.java]: The toString is meant to be the syntax element more or less how it appears in the script

**Comment by Blueyescat** [comment]
@Pikachu920 What do you mean? The warning names should be in english.lang.

**Comment by bensku** [comment]
Doesn't build, cannot merge.
